### ISSUE TITLE
[Feature Request] Duplicate Avatar

### ISSUE THREAD

**Issue by rurre** [jangxx/VRC-Avatar-Remote-Server#7]
I'd like to be able to duplicate and rename avatars inside a board without having to do it inside the config.
This would allow me to have the same controls, or almost the same controls on avatars that have a very similar setup.

**Comment by jangxx** [comment]
Renaming an avatar should be very straight-forward to implement, but I'm not sure how to go about the duplication part, especially since there isn't currently a way to edit the parameters behind controls at all or to have controls be in a "disabled" state with invalid options.

The only sensible way I can think of, would be to kind of match the parameter names and types between the avatars and discarding the rest, but that could still lead to situations where you would have to copy and edit controls within the config file.

I actually don't think doing these sorts of operations in the config file is even a bad thing in the first place. There is already no way to change every single config option in the UI and doing advanced things like duplicating avatars is _a lot_ simpler to do directly in the config file than creating a bunch of UI and frontend logic that would rarely get used. The whole reason I opted for YAML as the config file format instead of JSON is that it's easier to read and edit, since you don't have to think about commas and lists when copy-pasting stuff around and you also get nice readable indentation for free.

**Comment by rurre** [comment]
I assumed i'd have to copy paste the whole section, rename the avatar, change the avatar id, then replace all guids for all controls with new ones?

**Comment by jangxx** [comment]
You don't actually need to replace the control ids - they only need to be unique per avatar, so copy-pasting them doesn't cause any issues.

What you _do_ need to do are these steps:

- Copy the whole section below the `boards.<id>.avatars.<avid>` key
  - Change the avatar id
  - You can change the name if you want to, but you don't _have_ to change it
- Copy the whole section below the top-level `avatars.<avid>` key
  - Change the avatar id here as well
  - Make sure that the parameters are all pointing to the correct OSC addresses

The last part is crucial, since the server only listens and sends on addresses defined there. Normally these parameters are created every time a new control is added, but if you're creating new avatar controls directly in the config file you need to update this section as well, otherwise the board will not work.

**Comment by rurre** [comment]
Yeah, this ended up mostly working. Most toggles worked but not all, for some reason.

Dragging the the avatar json into the admin interface fixed it completely, for some reason.
________________________________

**Comment by jangxx** [comment]
>Actually, what fixed it is me resetting the osc config ingame (then putting the json in). I didn't realize I have to reset the config ingame.

Yes, if you change parameters on an avatar you always need to reset the OSC config before the changes are reflected in the OSC interface (and the OSC config file by extension).
